File: week_2/src/sensor_pkg/scripts/sensor_node.py
#!/usr/bin/env python3
#conding = utf-8

import rospy
from sensor_msgs.msg import LaserScan

def sensor_callback(msg):
    fdist = msg.ranges[180]
    rospy.loginfo("distant:",fdist)

if __name__ == "__main__":
    rospy.init_node("sensor_node")

    sub = rospy.Subscriber("/scan",LaserScan,sensor_callback,queue_size=10)

    rospy.spin()

File: week_2/src/sensor_pkg/src/sensor_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>

void SensorCallback(sensor_msgs::LaserScan msg)
{
    float fMidDist = msg.ranges[180];
    ROS_INFO("distant:%f",fMidDist);

}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc,argv,"sensor_node");

    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("/scan",10,SensorCallback);

    ros::spin();
    
    return 0;
}
